components/List/tests/unit/MDCSelfSizingStereoCellTests.m

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialList.h"
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTypography.h"
+#import "MaterialTypographyScheme.h"
+
+@interface MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow : UIWindow
+
+/** Used to override the value of @c preferredContentSizeCategory. */
+@property(nonatomic, copy) UIContentSizeCategory contentSizeCategoryOverride;
+
+@end
+
+@implementation MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = UIContentSizeCategoryLarge;
+  }
+  return self;
+}
+
+- (instancetype)initWithContentSizeCategoryOverride:
+    (UIContentSizeCategory)contentSizeCategoryOverride {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = contentSizeCategoryOverride;
+  }
+  return self;
+}
+
+- (UITraitCollection *)traitCollection {
+  if (@available(iOS 10.0, *)) {
+    UITraitCollection *traitCollection = [UITraitCollection
+        traitCollectionWithPreferredContentSizeCategory:self.contentSizeCategoryOverride];
+    return traitCollection;
+  }
+  return [super traitCollection];
+}
+
+@end
+
+@interface MDCSelfSizingStereoCellTests : XCTestCase
+
+@end
+
+@implementation MDCSelfSizingStereoCellTests
+
+- (void)testSelfSizingStereoCellDynamicTypeBehavior {
```

Comment:
`testContentSizeCategoryNotificationCausesFontsToAdjustOniOS10AndUp`

- You don't need to repeat the name of the class in the test.
- Please give unit test names descriptive names that indicate what the test's expectations are and the specific behavior it's testing.

Reply:
We may also want to document that the reason the behavior being tested here only works on iOS 10 and up is because we aren't able to simulate the UIApplication logical path for iOS 9, not because we don't support Dynamic Type on iOS 9.

Reply:
Addressed in 06ed43ca5795f72487a0b36d297c6b1c7a1b2a51.